packages/expect/src/utils.js

```diff
@@ -251,7 +251,9 @@ export const sparseArrayEquality = (a: any, b: any) => {
   // A sparse array [, , 1] will have keys ["2"] whereas [undefined, undefined, 1] will have keys ["0", "1", "2"]
   const aKeys = Object.keys(a);
   const bKeys = Object.keys(b);
-  return equals(a, b) && equals(aKeys, bKeys);
+  return (
+    equals(a, b, [iterableEquality, typeEquality], true) && equals(aKeys, bKeys)
```

Comment:
Should this have `sparseArrayEquality` too? There could be nested arrays for which we need to do that check. Or does that already work this way?

Reply:
It would create an infinite loop:
`toStrictEqual(a, b)` -> `equals(a, b, [iterableEquality, typeEquality, sparseArrayEquality], true)` -> `sparseArrayEquality(a, b)` -> `equals(a, b, [iterableEquality, typeEquality, sparseArrayEquality], true)` -> `...`

😁 I did try it at the beginning

Reply:
I've just checked and it also works e.g. for
```js
expect([[{a: undefined}]]).not.toStrictEqual([[]]);
expect([{a: [{a: undefined}]}]).not.toStrictEqual([{a: []}]);
```
I just wasn't quite sure how exactly the recursion works in the beginning.
Maybe add those test cases as well for peace of mind?

Reply:
I just added an extra test to check that strictEqual will work fine with deeper objects

Reply:
The regression and the question about correctness of its correction is evidence that as we continue to fix edge cases in `expect` package, we also need to think how to reduce the complexity of coupling between custom testers.